### PR TITLE
Replace all fuel that spawns in military bases with jp8

### PR DESCRIPTION
**Pull request by hexagonrecursion**
#### Summary

SUMMARY: Content "Switch fuel pumps in military bases from diesel to JP8"

#### Purpose of change

Fixes: #46746. Many of the cars inside of a mil base are filled with jp8 by default, but the pump only has diesel.

#### Describe the solution

Replace the fuel in the pump and jerrycans that spawn in a mil base with jp8.

#### Describe alternatives you've considered

Leave as is

#### Testing

1. The game loads
2. The pump in a military base contains jp8

#### Additional context

~Currently blocked by this:~ no longer blocked
https://github.com/CleverRaven/Cataclysm-DDA/blob/941d5fdfccf666b00afd0616cc1a4cd5ccb3f12a/src/mapgen.cpp#L1091-L1094

### CODE REVIEW

**Comment by Cyprex** [comment]
Note that (one of the?) military helipads have a warehouse with 100L steel drums, of the ones that were filled I found 2 with JP8, one with diesel and one with gasoline, I guess this is nice to have?

**Comment by actual-nh** [comment]
The above "Update mapgen.cpp" is to, well, hopefully fix the C++ code issue.

**Comment by hexagonrecursion** [comment]
@actual-nh 

I tried it out. Now I get an error when visiting a military base (see below) and instead of a pump I get a puddle of jp8 where the pump should be.

 DEBUG    : invalid terrain id ""

 FUNCTION : int_id<T> generic_factory<ter_t>::convert(const string_id<T> &, const int_id<T> &, const bool) const [T = ter_t]
 FILE     : src/generic_factory.h
 LINE     : 467

**Comment by hexagonrecursion** [comment]
I think I figured out why this happens: jp8 is missing pump_terrain:
https://github.com/CleverRaven/Cataclysm-DDA/blob/a149ceb00807a323cd7eba5fa7a3a7c8b3b0b1b2/data/json/materials.json#L755-L775

**Comment by actual-nh** [comment]
Looks like a new "t_jp8_pump" terrain is needed - probably just copy from the diesel variety.

**Comment by hexagonrecursion** [comment]
> Looks like a new "t_jp8_pump" terrain is needed - probably just copy from the diesel variety.

Done. Will push in a moment.

File: data/json/furniture_and_terrain/terrain-manufactured.json [hunk 1]
```diff
@@ -156,6 +156,54 @@
       ]
     }
   },
+  {
+    "type": "terrain",
+    "id": "t_jp8_pump",
+    "name": "JP8 pump",
+    "looks_like": "t_gas_pump",
+    "description": "This pump is filled with JP8. If this gas dispenser doesn't give up the goods for free, you may have to pay at a nearby terminal.",
```
Comment (actual-nh) [inline data/json/furniture_and_terrain/terrain-manufactured.json:164]:
```suggestion
    "description": "This pump is filled with JP8.  If this gas dispenser doesn't give up the goods for free, you may have to pay at a nearby terminal.",
```
Is this actually true of military bases and pumps meant for military vehicles, BTW (the charging)?

Comment (hexagonrecursion) [reply data/json/furniture_and_terrain/terrain-manufactured.json]: > Is this actually true of military bases and pumps meant for military vehicles, BTW (the charging)?

It does not look like it is. I was thinking this pump might at some point be added in other places.

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/no-hope-mod-v2-1/24971/22

**Pull request merged**